### PR TITLE
[cosmic-swingset] Add unit tests for lib-wallet

### LINKED ISSUE #1051: [Cosmic-swingset] Add unit test for wallet.cancelOffer to test-lib-wallet.js
Most unit tests are added in #1056 

Still to be done:

- [ ] Test `wallet.cancelOffer`

### PR DESCRIPTION
**Pull request by katelynsills**
Does most of #1051

Adds unit tests for lib-wallet.

The one thing I haven't written unit tests for is `wallet.cancelOffer` which is much more intensive since I can't properly test it with automaticRefund. I would need a contract that holds onto offers like simpleExchange. I think I would prefer doing it in another PR so these tests could land.

### CODE REVIEW

File: packages/cosmic-swingset/test/unitTests/test-lib-wallet.js [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+import produceIssuer from '@agoric/ertp';
+import { makeZoe } from '@agoric/zoe';
+import { makeRegistrar } from '@agoric/registrar';
+import harden from '@agoric/harden';
+import { makeGetInstanceHandle } from '@agoric/zoe/src/clientSupport';
+
+import { makeWallet } from '../../lib/ag-solo/vats/lib-wallet';
+
+const setupTest = async () => {
+  const contractRoot = `${__dirname}/automaticRefund`;
```
Comment (michaelfig) [inline packages/cosmic-swingset/test/unitTests/test-lib-wallet.js]: As you asked, the answer is "yes".  You can use `require.resolve('@agoric/zoe/src/contracts/automaticRefund')` to find the path to it without the trampoline module.

File: packages/cosmic-swingset/test/unitTests/test-lib-wallet.js [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+            },
+            exit: { onDemand: null },
+          },
+          requestContext: { origin: 'unknown' },
+          status: undefined,
+        },
+      ],
+      `offer structure`,
+    );
+    const { outcome } = await wallet.acceptOffer(id);
```
Comment (michaelfig) [inline packages/cosmic-swingset/test/unitTests/test-lib-wallet.js]: The wallet used to return `depositedP` which allowed you to synchronise with the actual deposit completion.  It seems that functionality has been lost.  There is no way to know if the spoils have been deposited, so I think your test wins the race before that happens.

**Review by michaelfig — APPROVED** [review]
Looks good.  `lib-wallet.js` `acceptOffer` just needs to expose `depositedP` to be usable programmatically.

File: packages/cosmic-swingset/test/unitTests/test-lib-wallet.js (hunk 1 above)
Comment (katelynsills) [reply packages/cosmic-swingset/test/unitTests/test-lib-wallet.js]: Thanks! I'll do that

File: packages/cosmic-swingset/test/unitTests/test-lib-wallet.js (hunk 2 above)
Comment (katelynsills) [reply packages/cosmic-swingset/test/unitTests/test-lib-wallet.js]: Gotcha, I must have removed that. I'll add it back. thanks!

Comment (katelynsills) [reply packages/cosmic-swingset/test/unitTests/test-lib-wallet.js]: That's added back, and I did a manual test with dapp-encouragement as well

**Pull request merged**